From e6fc6072ebe5c14c6e6d5efe1a21387c27758962 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Wed, 24 Aug 2022 10:11:52 +0300
Subject: [PATCH 089/102] dt-bindings: gpio: update SIUL2 gpio bindings

Update the dt-bindings for the GPIO driver to be similar with the ones
from Linux.

Issue: ALB-9214

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 doc/device-tree-bindings/gpio/gpio-s32cc.txt  |  27 -----
 doc/device-tree-bindings/gpio/gpio-s32cc.yaml | 100 ++++++++++++++++++
 2 files changed, 100 insertions(+), 27 deletions(-)
 delete mode 100644 doc/device-tree-bindings/gpio/gpio-s32cc.txt
 create mode 100644 doc/device-tree-bindings/gpio/gpio-s32cc.yaml

diff --git a/doc/device-tree-bindings/gpio/gpio-s32cc.txt b/doc/device-tree-bindings/gpio/gpio-s32cc.txt
deleted file mode 100644
index 2c2418179b..0000000000
--- a/doc/device-tree-bindings/gpio/gpio-s32cc.txt
+++ /dev/null
@@ -1,27 +0,0 @@
-* NXP S32CC SIUL2 GPIO controller
-
-Required properties:
-- compatible : "nxp,s32cc-siul2-gpio"
-- #gpio-cells : Should be 2 (pin number and his properties)
-- gpio-controller : Marks the device node as a GPIO controller
-- gpio-ranges: See gpio.txt
-
-Example:
-siul2_1@44010000 {
-	compatible = "simple-mfd";
-	status = "okay";
-	reg = <0x0 0x44010000 0x0 0x2000>;
-
-	pinctrl1: siul2-pinctrl1 {
-		...;
-	};
-
-	gpio1: siul2-gpio1 {
-		compatible = "nxp,s32cc-siul2-gpio";
-		#gpio-cells = <2>;
-		gpio-controller;
-		gpio-ranges = <&pinctrl1 0 112 11>,
-			<&pinctrl1 11 144 47>;
-		status = "okay";
-	};
-};
diff --git a/doc/device-tree-bindings/gpio/gpio-s32cc.yaml b/doc/device-tree-bindings/gpio/gpio-s32cc.yaml
new file mode 100644
index 0000000000..e4e76d408b
--- /dev/null
+++ b/doc/device-tree-bindings/gpio/gpio-s32cc.yaml
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: GPL-2.0
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/gpio/gpio-s32cc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: NXP S32CC SIUL2 GPIO controller
+
+maintainers:
+  - Radu Pirea <radu-nicolae.pirea@nxp.com>
+  - Andrei Stefanescu <andrei.stefanescu@nxp.com>
+
+description: |
+  Support for the SIUL2 GPIOs found on the S32CC chips.
+
+properties:
+  compatible:
+    items:
+      - enum:
+          - nxp,s32cc-siul2-gpio
+
+  reg:
+    maxItems: 6
+
+  reg-names:
+    items:
+      - const: opads0
+      - const: opads1
+      - const: ipads0
+      - const: ipads1
+      - const: eirqs
+      - const: eirq-imcrs
+
+  gpio-controller:
+    description:
+      Marks the device node as a gpio controller
+
+  "#gpio-cells":
+    description: |
+      Should be two.  The first cell is the pin number and
+      the second cell is used to specify the gpio polarity
+      0 = active high
+      1 = active low
+
+  gpio-ranges:
+    description:
+      Interaction with the PINCTRL subsystem
+
+  gpio-reserved-ranges:
+    description:
+      A list of start GPIO, number of GPIOs that are invalid if any such.
+
+patternProperties:
+  "^(hog-[0-9]+|.+-hog(-[0-9]+)?)$":
+    type: object
+    properties:
+      gpio-hog: true
+      gpios: true
+      input: true
+      output-high: true
+      output-low: true
+      line-name: true
+
+    required:
+      - gpio-hog
+      - gpios
+
+required:
+  - compatible
+  - reg
+  - reg-names
+  - gpio-controller
+  - "#gpio-cells"
+  - gpio-ranges
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/memory/s32-siul2.h>
+
+    gpio: siul2-gpio@4009d700 {
+      compatible = "nxp,s32g-siul2-gpio";
+      reg = <OPADS_SIUL2_0        0x10>,
+            <OPADS_SIUL2_1        0x18>,
+            <IPADS_SIUL2_0        0x10>,
+            <IPADS_SIUL2_1        0x18>,
+            <EIRQS_SIUL2_1        0xb4>,
+            <IMCRS_SIUL2_1_EIRQS  0x80>;
+      reg-names = "opads0", "opads1", "ipads0",
+                  "ipads1", "eirqs", "eirq-imcrs";
+      gpio-controller;
+      #gpio-cells = <2>;
+                    /* GPIO 0-101 */
+      gpio-ranges = <&pinctrl 0   0   102>,
+                    /* GPIO 112-190 */
+                    <&pinctrl 112 112 79>;
+      gpio-reserved-ranges = <102 10>, <123 21>;
+      status = "okay";
+    };
-- 
2.17.1

